src/main/java/com/google/sps/servlets/UserServlet.java

```diff
@@ -188,6 +200,30 @@ private void postHandleUnsave(List<Long> saved, long eventId) {
     for (int i = 0; i < saved.size(); i++) {
       if (saved.get(i) == eventId) {
         saved.remove(i);
+
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        Key eventKey = KeyFactory.createKey("Event", eventId);
+        try {
+          Entity eventEntity = datastore.get(eventKey);
+          Object attendees = eventEntity.getProperty("attendeeCount");
+          int attendeeCount = -1;
+          if (attendees != null) {
+            try {
+              attendeeCount += Integer.parseInt(attendees.toString());
+            } catch (NumberFormatException num) {
+              LOGGER.info("error parsing attendee count  for event id " + eventId);
+              attendeeCount = 0;
+            }
+          }
+          if (attendeeCount < 0) {
+            attendeeCount = 0;
+          }
```

Comment:
Would it be equivalent to initialize attendeeCount to 0, and delete these lines?

Reply:
I just wanted to account for the case where eventEntity.getProperty("attendeeCount") == 0 (hence attendeeCount == -1 without the check). It shouldn't ever happen because of the other checks in place, but I wanted to add it just in case.